great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1608,6 +1660,85 @@ def add_store(self, store_name: str, store_config: StoreConfigTypedDict) -> Stor
         self._save_project_config()
         return store
 
+    @public_api
+    @new_method_or_class(version="0.17.2")
+    def add_data_docs_site(
+        self, site_name: str, site_config: DataDocsSiteConfigTypedDict
+    ) -> None:
+        """Add a new Data Docs Site to the DataContext.
+
+        Example site config dicts can be found in our "Host and share Data Docs" guides.
+
+        Args:
+            site_name: New site name to add.
+            site_config: Config dict for the new site.
+        """
+        if self.config.data_docs_sites is not None:
+            if site_name in self.config.data_docs_sites:
+                raise gx_exceptions.InvalidKeyError(
+                    f"Data Docs Site `{site_name}` already exists in the Data Context."
+                )
+
+            sites = self.config.data_docs_sites
+            sites[site_name] = site_config
+            self.variables.data_docs_sites = sites
+            self._save_project_config()
+
+    @public_api
+    @new_method_or_class(version="0.17.2")
+    def list_data_docs_sites(
+        self,
+    ) -> dict:
```

Comment:
Can we be more specific at all? `dict[str, someObj]`?

Reply:
Good call, changed to `dict[str, DataDocsSiteConfigTypedDict]`

Reply:
I accidentally included in another branch instead of here, good catch! It is included now.